### PR TITLE
Add option to enable fast IDCT in libjpeg-turbo

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

### CODE REVIEW

**Comment by jantonguirao** [comment]
!build

File: dali/pipeline/operators/decoder/image_decoder.cc [hunk 1]
```diff
@@ -49,6 +49,11 @@ is encountered and internal buffer needs to be reallocated to decode it.)code",
 `batch_size*prefetch_queue_depth` during the construction and suballocate them
 in runtime. Ignored when `split_stages` is false.)code",
       false)
+  .AddOptionalArg("use_fast_dct",
+      R"code(Enables fast IDCT in libjpeg-turbo decompressor when the library is used (CPU JPEG decoding).
```
Comment (JanuszL) [inline dali/pipeline/operators/decoder/image_decoder.cc]:
```suggestion
      R"code(Enables fast IDCT in CPU based decompressor when the library is used (CPU JPEG decoding).
```
Add note that it applies when the GPU implementation fallback to CPU as well.

**Comment by JanuszL** [comment]
RN50 training perf and accuracy is not affected by this change. So removing usage of this option from RN50 training script.

**Comment by JanuszL** [comment]
!build

File: dali/image/image.h [hunk 2]
```diff
@@ -92,6 +92,15 @@ class Image {
     };
   }
 
+  inline void SetUseFastDct(bool use_fast_dct) {
```
Comment (mzient) [inline dali/image/image.h]: A general remark for this PR: Shouldn't it be "idct" everywhere? Or are we somehow preparing for JPEG _encoding_ and want to keep the flag name?

Comment (JanuszL) [reply dali/image/image.h]: Done

**Comment by JanuszL** [comment]
!build

**Pull request merged**